octree_web_viewer/src/backend.rs

```diff
@@ -65,82 +77,105 @@ fn pad(input: &mut Vec<u8>) {
+                // which requires careful constructing on the server and parsing on the client.
+                let mut reply_blob = Vec::<u8>::new();
+
+                let mut num_nodes_fetched = 0;
+                let mut num_points = 0;
+                for node_id in nodes_to_load {
+                    let mut node_data = octree
+                        .get_node_data(&node_id)
+                        .map_err(|_error| {
+                            crate::backend_error::PointsViewerError::NotFound(format!(
+                                "Could not get node {}.",
+                                node_id
+                            ))
+                        })
+                        .unwrap();
+
+                    // Write the bounding box information.
+                    let min = node_data.meta.bounding_cube.min();
+                    reply_blob.write_f32::<LittleEndian>(min.x).unwrap();
+                    reply_blob.write_f32::<LittleEndian>(min.y).unwrap();
+                    reply_blob.write_f32::<LittleEndian>(min.z).unwrap();
+                    reply_blob
+                        .write_f32::<LittleEndian>(node_data.meta.bounding_cube.edge_length())
+                        .unwrap();
+
+                    // Number of points.
+                    reply_blob
+                        .write_u32::<LittleEndian>(node_data.meta.num_points as u32)
+                        .unwrap();
+
+                    // Position encoding.
+                    let bytes_per_coordinate =
+                        node_data.meta.position_encoding.bytes_per_coordinate();
+                    reply_blob.write_u8(bytes_per_coordinate as u8).unwrap();
+                    assert!(
+                        bytes_per_coordinate * node_data.meta.num_points as usize * 3
+                            == node_data.position.len()
+                    );
+                    assert!(node_data.meta.num_points as usize * 3 == node_data.color.len());
+                    pad(&mut reply_blob);
+
+                    reply_blob.append(&mut node_data.position);
+                    pad(&mut reply_blob);
+
+                    reply_blob.append(&mut node_data.color);
+                    pad(&mut reply_blob);
+
+                    num_nodes_fetched += 1;
+                    num_points += node_data.meta.num_points;
+                }
+
+                let duration_ms = (time::precise_time_ns() - start) as f32 / 1000000.;
+                println!(
+                    "Got {} nodes with {} points ({}ms).",
+                    num_nodes_fetched, num_points, duration_ms
+                );
+
+                result(Ok(HttpResponse::Ok()
+                    .content_type("application/octet-stream")
+                    //disabling default encoding:
+                    //same data and speed of the iron implementation (testing performances on the same machine for client and server)
```

Comment:
`iron`? I thought you tested it for actix...

We usually put a space between `//` and the comment text, e.g. `// disabling...` instead of `//disabling...`

Reply:
I still would have preferred to include a `TODO` here that we need to test whether this makes sense in production...

Reply:
ok! I did not know ho to parameter the performance so I have just compared it with the iron implementation on the local machine. Todo inserted

Reply:
i've just pushed the comment changes  👀feuerste